### ISSUE TITLE
Ruby implementation

### ISSUE THREAD

**Issue by bracketttc** [badges/shields#753]
I would like to create a Ruby Gem implementation that could be deployed on private server instances, even offline instances.

My end goal is integration into [Gitlab](http://gitlab.com) and Gitlab CI.

Related to:
#541 
[Gitlab CE &#35;19352](https://gitlab.com/gitlab-org/gitlab-ce/issues/19352)

**Comment by espadrine** [comment]
If your goal is only to offer Gitlab / Gitlab CI integrations, I recommend that you not worry about the tricky bits of image generation, and simply focus on computing the badge's text.

Then, you will be able to download a badge with the format `https://img.shields.io/badge/:left-:right-:color.svg`.

What do you think?

**Comment by bracketttc** [comment]
I think that that would be a very simple and viable solution for most use-cases.

I would still like to develop a Ruby implementation that does all of the `tricky bits of image generation`.

**Comment by espadrine** [comment]
glhf

Shields' secret weapon in this regard is https://www.npmjs.com/package/pdfkit.

**Comment by bogdanRada** [comment]
I might be wrong, but i don't think this can be done since the license for this repo doesn't allow copying the svg templates. 

And in order to create a ruby gem you'll have to duplicate that content and also the logic for calculating the widths of text. Maybe i am wrong.

**Comment by bogdanRada** [comment]
The reason why i am asking about the license, is because honestly i started working on this, and i have a very good version ( which i initially made it for this https://github.com/bogdanRada/ruby-gem-downloads-badge   - in order to be able to serve badges even if shields.io service is down, which would have been activated only in that particular case ) 

and today i wanted to release it , but after reading the license i was confused, and i had to reset all the history and re-create the repository with the changes reverted.

The implementation is now private on my local computer . I would really appreciate if someone could explain what this license that there is on this repository currently allows. 

Basically i want to know if the license allows copying the svg templates and the logic for calculating the width. The implementation is purely Ruby though ( with the help of some other gems) 

If this is not allowed by the license, i will simply destroy all the changes i have made locally . But wanted to make sure before i do that.

**Comment by bracketttc** [comment]
[CC0](LICENSE.md) gives up as many of the copyright/licensing rights as possible under whatever jurisdictions are involved. The templates and code are committed to the public domain and so you, I, or anyone interested may use the templates to create their own implementations.

**Comment by bogdanRada** [comment]
awesome :) Thanks. I was planning to release a ruby implementation for this repo https://github.com/bogdanRada/ruby-gem-downloads-badge and i wasn;t sure the license allows that, so i reverted all changes .

By default the application would still rely on shields.io and only use that implementation if shields.io is unavailable.

Thank you very much. I was really worried that i might have commited something that was not allowed. Thanks a lot.

**Comment by espadrine** [comment]
I concur with @bracketttc — I specifically wanted the code and the assets to be maximally reusable. The goal is aesthetic consistency and quality whatever the means.

To be more exhaustive, however, the Verdana font itself is not under a free license, and cannot even be published. It is relied upon for accurate text width computation — more specifically, the one bundled with Macs, not the one that Windows or Linux package managers offer, which are distinct.

Also, all assets under logo/ are trademark of their respective companies and are under their terms and license.

**Comment by bogdanRada** [comment]
Thank you for detailed explanation . I don;t use the images under /logo directory at all, i only use the  files unde the /templates directory, and i implemented same calculations you have in NodeJS in Ruby also, 

However about the font Verdana,  when you say it can not be published you mean that i can not host it on a application? i need to use a link instead to google fonts website? Or does it mean that i need to pay for a license? I am a bit confuse and i am not very familiar with all the legal aspects related to fonts.

**Comment by bracketttc** [comment]
You can't distribute Verdana. It can't be in your gem.

If you have it on your client or server machine that is creating the badges, that's fine. An SVG isn't rendered on the server-side, but on the client-side, so a client will only see the badge in Verdana if they have Verdana.

**Comment by bogdanRada** [comment]
I was using it on the server only , but only to measure the width of a string, but just to be on the safe side, i reverted the changes on the server.  This way i will avoid all possible issues. I will just stick to using shields.io instead though. Thank you very much for the help. I had no idea this "free" fonts are so complicated to be used :(

**Comment by espadrine** [comment]
I'll close this, as there is nothing actionable in the Shields project. @bracketttc, when you have a working repo, we can transfer it to the badges organization (as are other badge-related projects) if you want. Ping me @espadrine on Twitter for that.
